src/audio/crossover/crossover.c

```diff
@@ -30,13 +28,16 @@
 #include <ipc/stream.h>
 #include <ipc/topology.h>
 #include <user/trace.h>
-#include <user/crossover.h>
 #include <user/eq.h>
 #include <errno.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <limits.h>
 
+#include "crossover_algorithm.h"
+#include "crossover_user.h"
```

Comment:
is there a doc/comment/anything explaining the new expected layout of the headers?

Reply:
@cujomalainey  I was looking for same here -> please see  https://github.com/thesofproject/sof/pull/8557#discussion_r1414487762 explanation from @lgirdwood 

Let me file an enhancement retrospectively -- these cross-module usages require alignment for sure.

Reply:
Ok what about how modules that depend on each other, is the plan to just to reach into their private files?

Reply:
@cujomalainey ，this is the GH link: https://github.com/thesofproject/sof/issues/8630
we may need discuss how to implement modules that depend on each other.